dd-sdk-android/src/main/kotlin/com/datadog/android/DatadogEventListener.kt

```diff
@@ -0,0 +1,172 @@
+        protocol: Protocol?
+    ) {
+        super.connectEnd(call, inetSocketAddress, proxy, protocol)
+        connend = System.nanoTime()
+    }
+
+    /** @inheritdoc */
+    override fun secureConnectStart(call: Call) {
+        super.secureConnectStart(call)
+        sslStart = System.nanoTime()
+    }
+
+    /** @inheritdoc */
+    override fun secureConnectEnd(call: Call, handshake: Handshake?) {
+        super.secureConnectEnd(call, handshake)
+        sslEnd = System.nanoTime()
+    }
+
+    /** @inheritdoc */
+    override fun responseHeadersStart(call: Call) {
+        super.responseHeadersStart(call)
+        headersStart = System.nanoTime()
+    }
+
+    /** @inheritdoc */
+    override fun responseHeadersEnd(call: Call, response: Response) {
+        super.responseHeadersEnd(call, response)
+        headersEnd = System.nanoTime()
+    }
+
+    /** @inheritdoc */
+    override fun responseBodyStart(call: Call) {
+        super.responseBodyStart(call)
+        bodyStart = System.nanoTime()
+    }
+
+    /** @inheritdoc */
+    override fun responseBodyEnd(call: Call, byteCount: Long) {
+        super.responseBodyEnd(call, byteCount)
+        bodyEnd = System.nanoTime()
+
+        val key = identifyRequest(call.request())
+        val timing = buildTiming()
+        (GlobalRum.get() as? DatadogRumMonitor)?.addResourceTiming(key, timing)
+    }
+
+    // endregion
+
+    // region Internal
+
+    private fun buildTiming(): RumEventData.Resource.Timing {
+        val (dnsS, dnsD) = if (dnsStart == 0L) {
+            0L to 0L
+        } else {
+            (dnsStart - callStart) to (dnsEnd - dnsStart)
+        }
+        val (conS, conD) = if (connStart == 0L) {
+            0L to 0L
+        } else {
+            (connStart - callStart) to (connend - connStart)
+        }
+        val (sslS, sslD) = if (sslStart == 0L) {
+            0L to 0L
+        } else (sslStart - callStart) to (sslEnd - sslStart)
+        val (fbS, fbD) = if (headersStart == 0L) {
+            0L to 0L
+        } else {
+            (headersStart - callStart) to (headersEnd - headersStart)
+        }
+        val (dlS, dlD) = if (bodyStart == 0L) {
+            0L to 0L
+        } else {
+            (bodyStart - callStart) to (bodyEnd - bodyStart)
+        }
+
+        return RumEventData.Resource.Timing(
+            dnsStart = dnsS,
+            dnsDuration = dnsD,
+            connectStart = conS,
+            connectDuration = conD,
+            sslStart = sslS,
+            sslDuration = sslD,
+            firstByteStart = fbS,
+            firstByteDuration = fbD,
+            downloadStart = dlS,
+            downloadDuration = dlD
+        )
+    }
+
+    /**
+     * Datadog's RUM implementation of OkHttp [EventListener.Factory].
+     * Adding this Factory to your [OkHttpClient] will allow Datadog to monitor
+     * timing information for your requests (DNS resolution, time to first byte, …).
+     */
+    class Factory : EventListener.Factory {
```

Comment:
Maybe we should add an example also in the *docs* on how to use this.

Reply:
Yes indeed

Reply:
Done